pkg/server/endpoints/harvester.go

```diff
@@ -59,42 +158,44 @@ func (h *HarvesterAPIHandlers) Onboard(echoCtx echo.Context, params harvester.On
 
 	if params.JoinToken == "" {
 		err := errors.New("join token is required")
-		return h.handleErrorAndLog(err, err, http.StatusBadRequest)
+		return h.handleErrorAndLog(err, err.Error(), http.StatusBadRequest)
 	}
 
 	token, err := h.Datastore.FindJoinToken(ctx, params.JoinToken)
 	if err != nil {
-		msg := errors.New("error looking up token")
+		msg := "error looking up token"
+		err := fmt.Errorf("%s: %w", msg, err)
 		return h.handleErrorAndLog(err, msg, http.StatusBadRequest)
 	}
 
 	if token == nil {
 		err := errors.New("token not found")
-		return h.handleErrorAndLog(err, err, http.StatusBadRequest)
+		return h.handleErrorAndLog(err, err.Error(), http.StatusBadRequest)
 	}
 
 	if token.ExpiresAt.Before(time.Now()) {
-		err := fmt.Errorf("token expired: trust domain ID: %s", token.TrustDomainID)
-		msg := fmt.Errorf("token expired")
+		msg := "token expired"
+		err := fmt.Errorf("%s: trust domain ID: %s", msg, token.TrustDomainID)
 		return h.handleErrorAndLog(err, msg, http.StatusBadRequest)
```

Comment:
Shouldn’t we use 401 instead of 400 ?
```suggestion
		return h.handleErrorAndLog(err, msg, http.StatusUnauthorized)
```

Reply:
Both are valid options. In my perspective `http.StatusBadRequest` is appropriate because the expiration of the token is a client-side error. The client should have provided a valid and non-expired token.

Reply:
Don’t you think this can also lead to some misleading interpretation of the error regarding the request body like malformed syntax, etc..? with `http.StatusUnauthorized` it should make it easier to figure out that is something wrong with the token and not the syntax of the request. Or that’s crazy of me hehe ?